### PR TITLE
Add a Qgs3D singleton object, designed to collect other singleton-ish objects in a central place

### PR DESCRIPTION
**Pull request by nyalldawson**
Copying the pattern setup by the QgsGui singleton container

### CODE REVIEW

File: src/3d/qgs3d.cpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgs3d.h"
+
+Qgs3D *Qgs3D::instance()
+{
+  static Qgs3D *sInstance( new Qgs3D() );
+  return sInstance;
```
Comment (m-kuhn) [inline src/3d/qgs3d.cpp:23]: Should we use `Q_GLOBAL_STATIC` so it's cleaned up on exit?

Comment (nyalldawson) [reply src/3d/qgs3d.cpp:23]: Possibly? If we apply it to QgsGui too...

Comment (m-kuhn) [reply src/3d/qgs3d.cpp:23]: Worth a try, with a potential revert if we have crashes on exit because of interdependencies.

Comment (nyalldawson) [reply src/3d/qgs3d.cpp:23]: Done

**Comment by nyalldawson** [comment]
@m-kuhn Windows says no: https://cdash.orfeo-toolbox.org/testDetails.php?test=21270597&build=47599

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**